lib/install/planbuilder.go

```diff
@@ -555,19 +625,70 @@ func splitServers(servers []storage.Server, app app.Application) (masters []stor
 	return masters, nodes, nil
 }
 
-// skipDependency returns true if the dependency package specified by dep
-// should be skipped when installing the provided application
-func (b *PlanBuilder) skipDependency(dep loc.Locator) bool {
-	// rbac-app is installed separately
-	if dep.Name == constants.BootstrapConfigPackage {
-		return true
+func addResources(builder *PlanBuilder, resourceBytes []byte, runtimeResources []runtime.Object) error {
+	kubernetesResources, gravityResources, err := splitResources(resourceBytes)
+	if err != nil {
+		return trace.Wrap(err)
 	}
-	// do not install bandwagon unless the app uses it in its post-install
-	if dep.Name == defaults.BandwagonPackageName {
-		setup := b.Application.Manifest.SetupEndpoint()
-		if setup == nil || setup.ServiceName != defaults.BandwagonServiceName {
-			return true
+	if len(gravityResources) != 0 {
+		var buf bytes.Buffer
+		err = storage.Encode(gravityResources, &buf)
+		if err != nil {
+			return trace.Wrap(err)
 		}
+		builder.gravityResources = buf.Bytes()
 	}
-	return false
+	for _, res := range gravityResources {
+		if res.Kind != storage.KindRuntimeEnvironment {
+			continue
+		}
+		env, err := storage.UnmarshalEnvironmentVariables(res.Raw)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		builder.env = env.GetKeyValues()
+		configmap := opsservice.NewEnvironmentConfigMap(env.GetKeyValues())
+		kubernetesResources = append(kubernetesResources, configmap)
+	}
+	kubernetesResources = append(kubernetesResources, runtimeResources...)
+	if len(kubernetesResources) != 0 {
+		var buf bytes.Buffer
+		err = resources.NewResource(kubernetesResources...).Encode(&buf)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		builder.resources = buf.Bytes()
+	}
+	return nil
+}
+
+func splitResources(resourceBytes []byte) (kubernetesResources []runtime.Object, gravityResources []storage.UnknownResource, err error) {
+	reader := bytes.NewReader(resourceBytes)
+	decoder := yaml.NewYAMLOrJSONDecoder(reader, defaults.DecoderBufferSize)
+	for err == nil {
+		var resource storage.UnknownResource
+		err = decoder.Decode(&resource)
+		if err != nil {
+			break
+		}
+		kind := modules.Get().CanonicalKind(resource.Kind)
```

Comment:
You can just do

```go
resource.Kind = modules.Get().CanonicalKind(resource.Kind)
```
here, then you won't need `resource.Kind = kind` in the "else" (and it won't affect the first "if" branch).